### ISSUE TITLE
Remove NumPy as dependency?

### ISSUE THREAD

**Issue by gahjelle** [bsoyka/advent-of-code-ocr#30]
Hi @bsoyka 

Thanks for making this! It worked perfectly also for Day 13, 2021 :+1: 
I had some issues with installing the library since NumPy has been pinned to a version that's not compatible with Python 3.10 in the version available on PyPI.

I looked at the code and noticed that it doesn't rely significantly on NumPy so that it's possible to make it work quite effectively only using the standard library. For instance, by doing something like:

```
    input_text = input_text.replace(fill_pixel, "#").replace(empty_pixel, ".")
    array = input_text.split("\n")

    rows, cols = len(array), len(array[0])
    if any(len(row) != cols for row in array):
        raise ValueError("all rows should have the same number of columns")
    if rows != 6:
        raise ValueError("incorrect number of rows (expected 6)")

    indices = [slice(start, start + 4) for start in range(0, cols, 5)]
    result = [
        ALPHABET_6["\n".join(row[index] for row in array)] for index in indices
    ]

    return "".join(result)
```

Would you be interested in me adding a PR that removes NumPy as a dependency, making the library easier to use?

Best,
Geir Arne

**Comment by bsoyka** [comment]
That would be amazing! Please do, if you'd like.

**Comment by gahjelle** [comment]
Closed by #32
